recipes/protobuf/all/conanfile.py

```diff
@@ -142,7 +142,8 @@ def generate(self):
             runtime = msvc_runtime_flag(self)
             if not runtime:
                 runtime = self.settings.get_safe("compiler.runtime")
-            tc.cache_variables["protobuf_MSVC_STATIC_RUNTIME"] = "MT" in runtime
+            if runtime:
+                tc.cache_variables["protobuf_MSVC_STATIC_RUNTIME"] = "MT" in runtime
```

Comment:
this block should be something along the lines of the following

```
runtime = self.settings.get_safe("compiler.runtime")
if runtime:
    tc.cache_variables["protobuf_MSVC_STATIC_RUNTIME"] = runtime == "static"
```

in Conan 2, `MT` will never be part of the `compiler.runtime` setting) - it may be part of `msvc_runtime_flag` - but seems unnecessary 

(cc @uilianries , @AbrilRBS )

Reply:
good catch!